File: client/src/app/model/todo.ts
export type Todo = Todo.NoId & {
  id: Todo.Id
};

export module Todo {
  export type     Id = Identity<'Todo'> & number;
  export function Id(id: number): Id { return id as Id; }

  export interface NoId {
    name: string;
    content: string;
  }
}

File: app/db/models/Todo.scala
package net.syrup16g.todo.db.model

import java.time.LocalDateTime

case class Todo(
  id: Option[Long],
  userId: Long,
  name: String,
  content: String,
  createdAt: LocalDateTime = LocalDateTime.now,
  updatedAt: LocalDateTime = LocalDateTime.now
)

